src/main/scala/cognite/spark/v1/DefaultSource.scala

```diff
@@ -126,6 +126,7 @@ class DefaultSource
       schema: StructType): BaseRelation = {
 
     val resourceType = parameters.getOrElse("type", sys.error("Resource type must be specified"))
+    val assetSubtreeIds = parameters.get("assetSubtreeIds").map(_.split(",").toList)
```

Comment:
not completely sure about this 🤔, can `externalId` have ',' in it?

Reply:
hmm good question, i don't know :'( we can try creating some in transformations with comma. If so, would you consider only allowing a single root in the filtering?

Reply:
yes it can unfortunately, I just tried to create one and re-query it, and not only , but almost any special character is accepted
![image](https://user-images.githubusercontent.com/92026183/193254644-d8c70469-1bf8-4d4c-9139-f98932eea754.png)

Reply:
I don't think it matters, the user can always use the numeric InternalId to avoid this problem.

For that matter, an externalId with only digits is also valid, but you'd interpret it as internal id. Again, easy to workaround by using the actual internal id.

Reply:
that is a really good point @silvavelosa but we need to give users a conversion function from id to externalId? as in dataset_id udf

Reply:
I don't know if there's any precedence of passing lists as options, but this is making me uncomfortable. Seems very unsafe, and like it will lead to surprising and confusing errors (when external ids contain `,`). If we have to string split, can we use something more unusual? `||` maybe? The customer in https://cognitedata.atlassian.net/wiki/spaces/INT/pages/3721166999/Asset+subtrees+retrieval+on+transformations would be happy with just one asset subtree, do we have to support multiple ones? Yes, we can, but can we save it for another day when/if a customer requests it, if they ever do?

Reply:
Or, for that matter, turn it into a JSON array `["\"asset1\",\"asset2\","3",3]`, and parse. Then we would even be able to tell "3" and 3 apart properly. Maybe options can have array values in Spark, but I seem to recall they're all just converted to strings, even if `.option("overwrite", true)` works...

Reply:
yep, that's how it went:

- integers like `123`, `456` are a single internalId
- valid json arrays of integers or strings are multiple internal/external ids
-- `["vnakls","2314","vnkalvs"]` are 3 externalIds
-- `[123, 2314, 123156]` are 3 internalIds
- anything else is a single externalId